src/com/company/DateValidator.java

```diff
@@ -0,0 +1,17 @@
+package com.company;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class DateValidator implements Validator {
+    /**
+     * @param str - this string which we are to check for valid;
+     * @return true - valid string, false - non-valid string;
+     */
+    @Override
+    public boolean validate(String str) {
```

Comment:
Use another approatch for "notifying" about errors in the code, instead of  return "boolean"